libraries/standard/json/utest/json_utest.c

```diff
@@ -746,6 +773,24 @@ void test_JSON_Search_Illegal_Documents( void )
     char * outValue;
     size_t outValueLength;
 
+    jsonStatus = JSON_Search( WHITE_SPACE,
+                              WHITE_SPACE_LENGTH,
+                              COMPLETE_QUERY_KEY,
+                              COMPLETE_QUERY_KEY_LENGTH,
+                              JSON_QUERY_SEPARATOR[ 0 ],
+                              &outValue,
+                              &outValueLength );
+    TEST_ASSERT_EQUAL( JSONIllegalDocument, jsonStatus );
```

Comment:
Why is the JSON_Search returning different return values for invalid JSON documents? 
JSONIllegalDocument on this line and JSONNotFound on line 810

Reply:
@dan4thewin mentioned that cases like these are expected when the user does not call JSON_Validate before calling JSON_Search, and the docs will be updated to say so